src/ui/index.ts

```diff
@@ -102,27 +104,79 @@ export function getNodeId({ attributes }: UiNode) {
 }
 
 /**
- * Filters nodes by their groups.
+ * Return the node input attribute type
+ * In <input> elements we have a variety of types, such as text, password, email, etc.
+ * When the attribute is null or the `type` attribute is not present, we assume it has no defined type.
+ * @param attr
+ * @returns type of node
+ */
+export const getNodeInputType = (attr: UiNodeAttributes): string =>
+  attr && "type" in attr ? attr.type : ""
```

Comment:
You could write this more compact (since you don't care about the type anyways):
```suggestion
export const getNodeInputType = (attr: any): string => attr?.["type"] ?? ""
```